pkg/config/config.go

```diff
@@ -38,36 +37,58 @@ type Set struct {
 	Configs     []Config
 }
 
+type Sysctler func(name string, params ...string) (string, error)
 type SysctlConfig struct {
 	Key, Value, DefaultValue string
+	SysctlFunc               Sysctler
 }
 
+type RouteAdder func(route *netlink.Route) error
+type RouteDeler func(route *netlink.Route) error
 type IPRouteConfig struct {
-	Route netlink.Route
+	Route    netlink.Route
+	RouteAdd RouteAdder
+	RouteDel RouteDeler
 }
 
+type RuleAdder func(rule *netlink.Rule) error
+type RuleDeler func(rule *netlink.Rule) error
+type RuleLister func(family int) ([]netlink.Rule, error)
 type IPRuleConfig struct {
-	Rule netlink.Rule
+	Rule     netlink.Rule
+	RuleAdd  RuleAdder
+	RuleDel  RuleDeler
+	RuleList RuleLister
 }
 
 type IPTablesRuleSpec []string
 
+type IPTabler interface {
+	NewChain(table, chain string) error
+	ClearChain(table, chain string) error
+	DeleteChain(table, chain string) error
+	AppendUnique(table, chain string, rulespec ...string) error
+	Delete(table, chain string, rulespec ...string) error
+}
+
 type IPTablesChainSpec struct {
 	TableName, ChainName string
 	IsDefaultChain       bool // Is a System default chain, if yes, we won't delete it.
+	Ipt                  IPTabler
 }
 
 type IPTablesRuleConfig struct {
 	Spec      IPTablesChainSpec
 	RuleSpecs []IPTablesRuleSpec
+	Ipt       IPTabler
```

Comment:
for golang style guide, for acroym, all should be capital, e.g., Html should be HTML